pkg/services/syncer/config/config.go

```diff
@@ -14,12 +14,18 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package services
+package config
 
 import (
-	provv1 "github.com/rancher-sandbox/rancheros-operator/pkg/apis/rancheros.cattle.io/v1"
+	"github.com/rancher-sandbox/rancheros-operator/pkg/clients"
 )
 
-type syncer interface {
-	sync() ([]provv1.ManagedOSVersion, error)
+type Config struct {
+	Requeuer      chan interface{}
+	Clients       *clients.Clients
+	OperatorImage string
```

Comment:
I am understanding it correctly that this is rancheros-operator wide? I would have expected to see the image tied to an specific channel crd, so you can have multiple custom syncers (or single one parsing multiple pods output) defined with different images. I guess this could be one of the motivations to move to a controller that spawns the image based in new channels, so there could potentially be an image per channel and define the image there.

Are you having in mind something like that for the controller?

Reply:
so this is maybe confusing and needs some inline comment indeed, this operatorImage is the actual image of the operator running in the cluster. The operator will use that to re-start itself in the pod that fetches the JSON data

Reply:
The image used to spawn the container is defined in another place, that config is "generic" for all the syncers